### ISSUE TITLE
UC Distribution Process Emails (11/29 Sprint) (SS) (On PROD)

### ISSUE THREAD

**Issue by TracyLinder** [cityofaustin/atd-data-tech#13395]
### For the new UC Distribution process we are building, we need to create automated system emails for the various attempt types. 

**Here are the attempt types that will have an automated email sent by the system:**

### Redistribute All
  
Subject: Austin Utility Location  - PERMIT # , PROJECT NAME (`Folder Name)`

Hello Jaclyn Lozano `[first, last name of People record]`

Thank you for your submittal to the Austin Utility Location and Coordination Committee (AULCC). Your Utility Coordination Application has been scheduled.

 AULCC Meeting date: 26-SEP-2023 `[AULCC Meeting date from process info tab]`

Meeting Location: **Online via Microsoft Teams** Contacts listed below will receive email invitation with Teams link/optional call-in number prior to meeting date.

Meeting Starts: 2:00 PM

 Please review your information.

If any data needs to be revised, added, or deleted; Please contact: [AULCC@austintexas.gov](mailto:AULCC@austintexas.gov)

AULCC Tracking Number #:UCC-123-123 `Folder info Austin Utility Coordination Tracking # (UCC/DUC #)`                                                ROWID: 13200840 `folderrsn`         

Project Name: Updating Emails Test App 1

Project Description: Test App 1 hello world hello world hello world hello world hello world hello world hello world hello world hello world hello world hello world hello world hello world hello world hello world hello world hello world hello world hello world hello world hello world hello world hello world

 

Design Stage: 90%   `(Folder Info 76106)`   Construction Start Date: 12/01/2025 `(Folder Info 76110)`        Construction End Date: 12/01/2030 `(Folder Info 76115)`

 

Applicant:  Jaclyn Lozano                                              E-Mail: jackie@atx.gov [if no email available display 'NO EMAIL']

Project Contact:  Jaclyn Lozano                                   E-Mail: NO EMAIL [if no email available display 'NO EMAIL']

Project Engineer: Jaclyn Lozano                                 E-Mail: NO EMAIL [if no email available display 'NO EMAIL']

Project Manager: Jaclyn Lozano                                 E-Mail: jackie@atx.gov [if no email available display 'NO EMAIL']

 

 

Once the Project is reviewed, please begin a Spreadsheet or Minutes Sheet to record responses, identify areas of conflict, and note proposed resolutions.

Projects that have been through the AULCC previously, must contain all utility comments from previous reviews. The Minutes tab, or a Conflict/Resolution spreadsheet should be utilized for this effort.               
Final review and clearance will be contingent on final review of the Conflict/Resolution Data and concurrence from the Project Engineer, Project Manager, Project Owner, and the AULCC Chairman.                         
AULCC Clearance Letter will be granted to the Project once all conflicts have been identified, and resolution has been verified. This letter will be required for [General Permit](https://www.austintexas.gov/page/general-permits) or [Site Plan](https://www.austintexas.gov/siteplans#submit-your-application) submittal to the Development Services Department online.

----

### Redistribute No Franchise

Subject: Your Utility Coordination Case Has Been Redistributed – PERMIT # , PROJECT NAME `(Folder Name)`

Hello:  Jaclyn Lozano `[Applicant first & last]`

Thank you for your resubmittal to the Austin Utility Location and Coordination Committee (AULCC).

Your case has been redistributed for review.

The review cycle end date is: 09/26/2023 `[Review due date process info]`

If you receive all approvals, please use the SMARTSHEET `Link to the lookup table with the URL` link to fill out the Completeness Letter Request Form. 

The SMARTSHEET URL IS https://app.smartsheet.com/b/form/dbdacb600e5a46a9bde80d58af1fcf25

### Redistribute Rejected

Subject: Your Utility Coordination Case Has Been Redistributed – PERMIT #  , PROJECT NAME `(Folder Name)`

Hello:  Jaclyn Lozano `[Applicant first & last]`

Thank you for your resubmittal to the Austin Utility Location and Coordination Committee (AULCC).

Your case has been redistributed to reviewers who had previous comments.

The review cycle end date is: 09/26/2023 `[Review due date process info]`

If you receive all approvals, please use the SMARTSHEET `Link to the lookup table with the URL` link to fill out the Completeness Letter Request Form. 

The SMARTSHEET URL IS https://app.smartsheet.com/b/form/dbdacb600e5a46a9bde80d58af1fcf25

----

### Initial Distribution and Initial Distribution No Franchise- **NO EMAIL**

**Email should go to the following people types when they exist on the UC people tab:**
- Applicant
- Contact
- Project Manager
- Project Engineer 

**Email should be from the stampuser who saves the process attempt.**

**Comment by TracyLinder** [comment]
### Update
- Greg P. is leading the initial documentation effort for this. He is still working on it and hoping to have something to look at by 9/8/23.

**Comment by TracyLinder** [comment]
Per Subha, we can use a lookup table to hold a single url value in the valid lookup table.

**Comment by TracyLinder** [comment]
**UC TEAM NOTES 9/25/23**

- We'll need two distinct email templates
- 
**Template 1 (with meeting date)**
-  Redistribute All

**Template 1 (with review due date)**
-  Redistribute No Franchise
-  Redistribute Rejected

**Comment by TracyLinder** [comment]
### Update

- The lookup table URL feature will work. Still waiting on email templates. 
- Jaclyn put the emails together and distributed them internally for feedback. They are still pending internal feedback. 
- Next UC meeting is 10/30 - let's send out a reminder to the group if we don't hear back by 10/20

**Comment by TracyLinder** [comment]
**UC Distribution – Redistribute No Franchise**

Subject: Your Utility Coordination Case Has Been Redistributed – PERMIT #,PROJECT NAME

Hello:  Jaclyn Lozano [Applicant first & last]

Thank you for your resubmittal to the Austin Utility Location and Coordination Committee (AULCC).

Your case has been redistributed for review.

The review cycle end date is: 09/26/2023 [Review due date process info]

If you receive all approvals, please use the SMARTSHEET `Link to the lookup table with the URL` link to fill out the Completeness Letter Request Form. 

----


The SMARTSHEET URL IS https://app.smartsheet.com/b/form/dbdacb600e5a46a9bde80d58af1fcf25

**Comment by TracyLinder** [comment]
**UC Distribution – Redistribute Rejected**

Subject: Your Utility Coordination Case Has Been Redistributed – PERMIT #,PROJECT NAME

Hello:  Jaclyn Lozano [Applicant first & last]

Thank you for your resubmittal to the Austin Utility Location and Coordination Committee (AULCC).

Your case has been redistributed to reviewers who had previous comments.

The review cycle end date is: 09/26/2023 [Review due date process info]

If you receive all approvals, please use the SMARTSHEET `Link to the lookup table with the URL` link to fill out the Completeness Letter Request Form. 

----


The SMARTSHEET URL IS https://app.smartsheet.com/b/form/dbdacb600e5a46a9bde80d58af1fcf25

**Comment by TracyLinder** [comment]
Subject: Austin Utility Location  - PERMIT #, PROJECT NAME

Hello Jaclyn Lozano [first, last name]

Thank you for your submittal to the Austin Utility Location and Coordination Committee (AULCC). Your Utility Coordination Application has been scheduled.

 AULCC Meeting date: 26-SEP-2023 [AULCC Meeting date from info tab]

Meeting Location: **Online via Microsoft Teams** Contacts listed below will receive email invitation with Teams link/optional call-in number prior to meeting date.

Meeting Starts: 2:00 PM

 Please review your information.

If any data needs to be revised, added, or deleted; Please contact: [AULCC@austintexas.gov](mailto:AULCC@austintexas.gov)

 

AULCC Tracking Number #:UCC-123-123                                                ROWID: 13200840         

 

Project Name: Updating Emails Test App 1

Project Description: Test App 1 hello world hello world hello world hello world hello world hello world hello world hello world hello world hello world hello world hello world hello world hello world hello world hello world hello world hello world hello world hello world hello world hello world hello world

 

Design Stage: 90%            Construction Start Date: 12/01/2025        Construction End Date: 12/01/2030

 

Applicant:  Jaclyn Lozano                                              E-Mail: jackie@atx.gov [if no email available display 'NO EMAIL']

Project Contact:  Jaclyn Lozano                                   E-Mail: NO EMAIL [if no email available display 'NO EMAIL']

Project Engineer: Jaclyn Lozano                                 E-Mail: NO EMAIL [if no email available display 'NO EMAIL']

Project Manager: Jaclyn Lozano                                 E-Mail: jackie@atx.gov [if no email available display 'NO EMAIL']

 

 

Once the Project is reviewed, please begin a Spreadsheet or Minutes Sheet to record responses, identify areas of conflict, and note proposed resolutions.

Projects that have been through the AULCC previously, must contain all utility comments from previous reviews. The Minutes tab, or a Conflict/Resolution spreadsheet should be utilized for this effort.               
Final review and clearance will be contingent on final review of the Conflict/Resolution Data and concurrence from the Project Engineer, Project Manager, Project Owner, and the AULCC Chairman.                         
AULCC Clearance Letter will be granted to the Project once all conflicts have been identified, and resolution has been verified. This letter will be required for [General Permit](https://www.austintexas.gov/page/general-permits) or [Site Plan](https://www.austintexas.gov/siteplans#submit-your-application) submittal to the Development Services Department online.




 

General Permits

The General Permit Program is a specific type of site development review, permitting, and inspection process for City of Austin Departments and Franchised Utilities. General Permits allow infrastru...

**Comment by vschandramouli** [comment]
Updated AC with formatting changes requested by Tracy, moving to Backlog.

**Comment by Nadin-Nader** [comment]
@SubhaShroff this has been sent to the DevOps CM Team to be deployed to DEV under **SCTASK0376949**

**Comment by Nadin-Nader** [comment]
@SubhaShroff this has been deployed to DEV. Please UAT

**Comment by SubhaShroff** [comment]
@Nadin-Nader  I tested this on DEV. The email contents need corrections. Will be working on it.

**Comment by Lpriya-LIT** [comment]
@SubhaShroff Created Dev deployment request , SCTASK0379525

**Comment by Lpriya-LIT** [comment]
@SubhaShroff  Changes prompted to Dev , Please verify

**Comment by SubhaShroff** [comment]
Thank you @Lpriya-LIT.
I verified it on DEV, and the emails look good. Please raise TEST deployment request.
[13395-f262f0aacfd41521-6bd1dfd19e2ed63b- 7ce08fb29886d401587c817d14f090f1b65e8ce3 -TEST Deployment Plan.docx](https://github.com/cityofaustin/atd-data-tech/files/13735265/13395-f262f0aacfd41521-6bd1dfd19e2ed63b-.7ce08fb29886d401587c817d14f090f1b65e8ce3.-TEST.Deployment.Plan.docx)
Thanks.

**Comment by Lpriya-LIT** [comment]
@SubhaShroff Created Test deployment request - SCTASK0380378

**Comment by SubhaShroff** [comment]
Thanks @Lpriya-LIT.
@Nadin-Nader Though the ticket SCTASK0380378 looks closed 2 commits are not deployed in TEST. Conveyed to George and he will take a look and will update.

**Comment by SubhaShroff** [comment]
@Nadin-Nader  Tested it on TEST, its ready for UAT. Thanks.

**Comment by Nadin-Nader** [comment]
Sent email to Kyle, Jaclyn, Greg and Darren informing them that this is ready for UAT on TEST. Awaiting Response

**Comment by SubhaShroff** [comment]
@Nadin-Nader  we are waiting on UAT feedback on TEST.

**Comment by KyleErich** [comment]
![image](https://github.com/cityofaustin/atd-data-tech/assets/130776506/73ea53f5-c319-48fa-85dc-e75164696596)
One minor nitpick -- the email greeting on all of these new email templates captures not just the First and Last names, but also the Organization name. Based on the AC, I think we only want the First and Last names. If it's not possible to isolate the Organization name away from them, then I think we can deal with that, but otherwise let's see if we can just have the individual names.

The only other item involves the emails for Redistribute No Franchise and Redistribute Rejected. The AC indicates that the Smartsheet link is supposed to come in a line after the "If you receive all approvals..." line, but this is nowhere to be found on the email. It links the Smartsheet instead in the spot where the AC mentions there should be a lookup table. Is this intentional?
![image](https://github.com/cityofaustin/atd-data-tech/assets/130776506/0dd20d89-33f0-4060-98ed-7e33ec55a649)

**Comment by SubhaShroff** [comment]
@KyleErich Thank you for the feedback.
1. I can fix the greetings taking out the organization name. I just have a question based on my observation, if there is no person name just organization name in the people record, what is expected to display in the greetings?
2. About the Smartsheet link, the mention of 'lookup table' is to pick the smartsheet link from the lookup table. Its not to display as it is in AC. And the long link that's showing in your 2nd screenshot is the actual Smartsheet link that is dynamically picked from Lookup table(which is added in there) instead writing directly in the development code. 
And I can explain why the URL is not showing as it is given in AC i.e., https://app.smartsheet.com/b/form/dbdacb600e5a46a9bde80d58af1fcf25. Its because the email is sent to your COA inbox. COA mail server encloses any URL with some safety stuff. So, it displays that way. Otherwise, if we click on the link, it opens the intended URL.

Please let me know if that make sense.
CC: @Nadin-Nader

**Comment by KyleErich** [comment]
@SubhaShroff @Nadin-Nader 
Thanks for the clarifications, Subha!

1. Understood. I think if there is no first or last name provided on the People record, then it's okay for the greeting to just say "Hello," without naming anyone or anything. So if it's possible for it to remain blank, that would be okay with me.
2. Got it, that makes sense. Disregard my concern here, I think this suffices.

I'm going to approve the Test UAT on this one, as my tests were otherwise successful. You can make the change in my first point before promotion to Prod. I'm not too concerned about going through another round of Test UAT for that minor alteration.

**Comment by SubhaShroff** [comment]
@KyleErich, @Nadin-Nader 
On second thought, if we enclose the URL and give a hyperlink that would look neat, as shown below. If that sentence with hyperlink is fine without any change in the sentence, I will do the change. If any change is required, please suggest.
![image](https://github.com/cityofaustin/atd-data-tech/assets/51680715/e4b1abf1-bd9a-4098-8288-3238caf55853)

**Comment by KyleErich** [comment]
@SubhaShroff @Nadin-Nader I agree. Only change I would make is the word Smartsheet doesn't need to be in all caps, just the first letter. Thanks!

**Comment by SubhaShroff** [comment]
Thanks Kyle. 
@Nadin-Nader SCTASK0383912 is submitted for DEV deployment of the code changes to feedback fixes.
Thank you.

**Comment by Nadin-Nader** [comment]
Still waiting on CTM to deploy

**Comment by SubhaShroff** [comment]
@Nadin-Nader  DEV deployment is done and I tested. Name is not done as expected. I worked on it and send for another DEV deoployment request - SCTASK0386765.
Thanks Priya.

**Comment by SubhaShroff** [comment]
@Nadin-Nader  Tested the above deployment in DEV. It is working as expected. Please create a TEST deployment for the following  commits. Thanks.
5607e2a2a4800dd05f86e26ec63937ef4940add7
ec60f4756a0d0b831b21b265ff8721600e0e66a3
[13395-5607e2a2a4800dd05 - ec60f4756a0d0b831 - TEST Deployment Plan.docx](https://github.com/cityofaustin/atd-data-tech/files/14013092/13395-5607e2a2a4800dd05.-.ec60f4756a0d0b831.-.TEST.Deployment.Plan.docx)

**Comment by Nadin-Nader** [comment]
@SubhaShroff this has been sent to the DevOps Cm Team tot be deployed to TEST under **SCTASK0388096**

**Comment by SubhaShroff** [comment]
@Nadin-Nader  @KyleErich  The TEST deployments for the following are done based on the feedback. I tested those for RSN- 13205040. It is working as expected. Please UAT on TEST. Thank you.
1. Display only first, last name in the greetings. If there is no name if its just an organization, display just Hello in the greetings.
2.  Give a Hyperlink for the Smartsheet URL.

![image](https://github.com/cityofaustin/atd-data-tech/assets/51680715/0529c7dc-dc3b-4253-8701-f2a9cea34db6)

**Comment by SubhaShroff** [comment]
> @SubhaShroff @Nadin-Nader Thanks for the clarifications, Subha!
> 
> 1. Understood. I think if there is no first or last name provided on the People record, then it's okay for the greeting to just say "Hello," without naming anyone or anything. So if it's possible for it to remain blank, that would be okay with me.
> 2. Got it, that makes sense. Disregard my concern here, I think this suffices.
> 
> I'm going to approve the Test UAT on this one, as my tests were otherwise successful. You can make the change in my first point before promotion to Prod. I'm not too concerned about going through another round of Test UAT for that minor alteration.

@Nadin-Nader  @KyleErich  I just remembered that you had mentioned that you approved the TEST UAT and we can send this ticket for PROD deployment with fixing the greetings and hyperlink.

So, @Nadin-Nader  here is the PROD deployment plan. Please raise the deployment request. Thank you.
[13395- PROD - Deployment Plan.docx](https://github.com/cityofaustin/atd-data-tech/files/14041979/13395-.PROD.-.Deployment.Plan.docx)

**Comment by Nadin-Nader** [comment]
@SubhaShroff this has been sent to the DevOps CM Team to be deployed to PROD under **SCTASK0388880**

**Comment by Nadin-Nader** [comment]
@SubhaShroff this has been sent to the DevOps CM Team to be deployed to PROD with #12631 under **SCTASK0392131**

**Comment by Nadin-Nader** [comment]
@SubhaShroff this is now on PROD. I will let Kyle and Jaclyn know

**Comment by Nadin-Nader** [comment]
Waiting on response from UC Team or Kyle

**Comment by Nadin-Nader** [comment]
Still waiting on response

**Comment by KyleErich** [comment]
@Nadin-Nader UAT on PROD approved. We'll open a new ticket if any issues are noticed relating to this matter in the future.
